### ISSUE TITLE
Leash menu doesn't open from main menu

### ISSUE THREAD

**Issue by lickx** [lickx/opencollar-os#1]
It works with <prefix>leashmenu but not from the main menu

**Comment by WendyStarfall** [comment]
I'm not really familiar with OpenSim but maybe you have to express the logical "or" at https://github.com/lickx/opencollar-os/blob/6-os/src/collar/oc_leash.lsl#L464 with individual brackets.

```LSL
        if ((sMessage=="leashmenu") || (sMessage == "menu leash")){
            list lButtons;
            if (kMessageID != g_kWearer) lButtons += "Grab";// Only if not the wearer.
            else lButtons += [" "];
            if (g_kLeashedTo != NULL_KEY) {
                if (g_bFollowMode) lButtons += ["Unfollow"];
                else lButtons += ["Unleash"];
            }
```

If that won't work maybe you could also try to just swap ``"menu leash"`` at https://github.com/lickx/opencollar-os/blob/6-os/src/collar/oc_leash.lsl#L621 with ``"leashmenu"``. At one point we wanted to establish continuity for menu calls with ``"menu featurename"``. In Six most menus can be called this way.

PS: Awesome to see someone is doing a real port. I starred it a while ago but I'm afraid I won't be much help for OpenSim. Still I can peek in and learn some stuff myself. Feel free to poke me on SL anytime. :speak_no_evil:

**Comment by lickx** [comment]
Thanks I tried both suggestions without success. However I found the subdialog code in DIALOG_RESPONSE (coming from the main dialog) never runs because g_lMenuIDs will be empty (and iMenuIndex will be -1). I remember hacking around this in OC3 too.

Maybe I find a better solution.. for now commit 19337124f4745cff9f0ca4f048c8fbf6b6e03d29

And thanks about the porting I do my best but I'm still learning loads of new things through this complex project while trying to grasp how the code base works exactly. There are a lot of quirks to deal with in opensim LSL :)
